### DISCUSSION TITLE
是不是不支持nextjs 服务端渲染

### DISCUSSION THREAD

**Discussion by NNNNzs** [imzbf/md-editor-rt#90] [Q&A] [answered]
是不是不支持nextjs 服务端渲染

**Comment by imzbf** [comment]
支持啊，是遇到什么问题了么

**Reply by JingpengZhang** [reply]
我是在 nextjs 13 里面用，只使用预览组件，但是会报这个错误
TypeError: createContext only works in Client Components. Add the "use client" directive at the top of the file to use it. Read more: https://nextjs.org/docs/messages/context-in-server-component

**Reply by imzbf** [reply]
方便提供一下出现这个问题的环境吗，虽然文档上这样说了，但是我创建了一个基础的环境尝试，却没有类似的提示
<img width="1724" alt="image" src="https://github.com/imzbf/md-editor-rt/assets/37092006/d0980346-b972-4c22-ba3d-1f16eb9cfcd8">

**Reply by JingpengZhang** [reply]
<img width="1512" alt="1" src="https://github.com/imzbf/md-editor-rt/assets/88316389/72223df9-d5f2-4bff-884f-f0fc7b63be8a">
<img width="1512" alt="2" src="https://github.com/imzbf/md-editor-rt/assets/88316389/7ce3f769-7afa-4277-863c-a8b633df3583">
谢谢大佬这么晚还在解决问题

**Reply by imzbf** [reply]
OK，我再尝试一下

**Reply by JingpengZhang** [reply]
你好，我发现这个问题是在服务端组件下出现的，而服务端组件下不能使用 useState 等，可以尝试在新建页面下是否会出现此问题。

**Reply by imzbf** [reply]
找了下文档，nextjs的服务端组件的区别
<img width="568" alt="image" src="https://github.com/imzbf/md-editor-rt/assets/37092006/00a31dd1-4db3-4a09-82fa-ced8825b91a7">

然后在13中这样尝试了一下
<img width="1726" alt="image" src="https://github.com/imzbf/md-editor-rt/assets/37092006/304359aa-bcf0-4718-ac39-7c8ffaf83b7d">

这个问题可能需要开发者分离一下逻辑来解决。

**Reply by JingpengZhang** [reply, marked as answer]
我尝试了下在我的项目首页文件使用，还是会出现这个错误。
<img width="1512" alt="1" src="https://github.com/imzbf/md-editor-rt/assets/88316389/3b0790f1-68fb-4dba-94de-7676f9ae7078">
另外，我注意到我们创建项目的时候可能选择了不同的路由模式（我这边是 AppRoute），所以我尝试了重新创建一个工程进行测试，结果发现无论是在首页还是动态页面上使用 md-editor-rt 都没有出现上述的问题。
<img width="533" alt="weapp截图20230712114122" src="https://github.com/imzbf/md-editor-rt/assets/88316389/64ec393f-23ab-4eb4-8f11-404b5b3a4f1b">
<img width="1512" alt="2" src="https://github.com/imzbf/md-editor-rt/assets/88316389/6a9e25f3-580f-4d51-9925-c4bc27cc04c2">

**Comment by imzbf** [comment]
合理利用它的设定即可，既然服务端组件不能使用react相关的接口，就在客户端组件使用就不存在这类问题了，至于怎么拆分，还需要开发者自己平衡。
